web-ui/src/components/checkin/CheckinHistory.jsx

```diff
@@ -1,120 +1,120 @@
-import React, { useContext, useState } from "react";
+import React, { useContext, useEffect, useState } from "react";
+
+import {
+  UPDATE_CURRENT_CHECKIN,
+  UPDATE_CHECKINS,
+} from "../../context/AppContext";
 import { AppContext } from "../../context/AppContext";
-import { getCheckinByPdlId } from "../../api/checkins";
+import { updateCheckin } from "../../api/checkins";
 
-import Avatar from "@material-ui/core/Avatar";
 import ArrowBackIcon from "@material-ui/icons/ArrowBack";
 import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
+import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
 import DatePicker from "react-datepicker";
 import "react-datepicker/dist/react-datepicker.css";
-import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
-import { getMember } from "../../api/member";
 
 import "./Checkin.css";
 
-const CheckinsHistory = ({ setIndex }) => {
-  const { state } = useContext(AppContext);
-  const { userProfile } = state;
-  const { workEmail, role, id, pdlId } =
-    userProfile && userProfile.memberProfile ? userProfile.memberProfile : {};
-  const { imageUrl, name } = userProfile ? userProfile : {};
-  const [checkins, setCheckins] = useState([]);
-  const [checkinIndex, setCheckinIndex] = useState(0);
-  const [pdl, setPDL] = useState();
+const CheckinsHistory = ({ history }) => {
+  const { state, dispatch } = useContext(AppContext);
+  const { checkins, currentCheckin } = state;
+  const [index, setIndex] = useState(0);
 
-  // Get PDL's name
-  React.useEffect(() => {
-    async function getPDLName() {
-      if (pdlId) {
-        let res = await getMember(pdlId);
-        let pdlProfile =
-          res.payload.data && !res.error ? res.payload.data : undefined;
-        setPDL(pdlProfile ? pdlProfile.name : "");
-      }
+  useEffect(() => {
+    if (checkins.length) {
+      const { pathname } = document.location;
+      const [, , checkinid] = pathname.split("/");
```

Comment:
This seems error prone...What happens if someone deploys this under a different root path? I would use react-router to get this value: https://reactrouter.com/web/example/url-params